crates/dht/src/time.rs

```diff
@@ -0,0 +1,1051 @@
+//! Partition of time into slices.
+//!
+//! Provides an encapsulation for calculating time slices that is consistent when computed
+//! by different nodes. This is used to group Kitsune2 op data into time slices for combined hashes.
+//! Giving multiple nodes the same time slice boundaries allows them to compute the same combined
+//! hashes, and therefore effectively communicate which of their time slices match and which do not.
+//!
+//! A time slice is defined to be an interval of time that is closed (inclusive) at the start and
+//! exclusive at the end. That is, the time slice includes the start timestamp but not the end.
```

Comment:
Had to read this a few times but I'm bad with words. I found this clearer:

/// A time slice represents an interval [start, end] where:
/// - `start`: Included in the slice (inclusive bound)
/// - `end`: Not included in the slice (exclusive bound)

Reply:
The notation is good, I'd like to keep the initial definition in terms of open/closed. Which I'd actually failed to do in the first place. Then I've added an alternative description that is just this.

Does that read better?